### ISSUE TITLE
[BUG] NiftiLabelsMasker missing signal

### ISSUE THREAD

**Issue by andreifoldes** [nilearn/nilearn#4113]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Operating system

- [ ] Linux
- [ ] Mac
- [X] Windows

### Operating system version

- Windows 10

### Python version

- [ ] 3.12
- [ ] 3.11
- [X] 3.10
- [ ] 3.9
- [ ] 3.8

### nilearn version

0.10.2

### Expected behavior

I have a mask with certified 8 non-zero values and would expect 8 numbers from NiLabelsMasker, that is indeed the case for all but one of my participants:

```
 right_subfield_mask = f"xxx/hippunfold/sub-{subject_label}/anat/sub-{subject_label}_hemi-R_space-T1w_desc-subfields_atlas-bigbrain_dseg.nii.gz"
    
    tmp = nib.load(right_subfield_mask)
    tmp = tmp.get_fdata()
    # get unique values nonzero values
    unique_values = np.unique(tmp[tmp != 0])
    print(unique_values)
    # print unique values    
    if len(unique_values) != 8:
        # raise an actual warning using warnings.warn
        # print unique values
        print(unique_values)
        warnings.warn("WARNING: unique values in right_subfield_mask is not equal to 8")
    
    right_subfield_labels = [ label + "_right" for label in subfield_labels]
    right_subfield_labels = [right_subfield_labels[i-1] for i in unique_values.astype(int)]

    # Instantiate the masker with label image and label values
    masker = NiftiLabelsMasker(
        right_subfield_mask,
        labels=right_subfield_labels,
        standardize="zscore_sample",
    )

    right_subfield_signals = masker.fit_transform(betamap)
    print(right_subfield_signals)
˙˙˙˙

### Current behavior & error messages

This is what I got:

8 non-zero values go in, 7 values come out. No warning or error message...

```bash
>> right_subfield_mask = f"/xxx/hippunfold_v2/hippunfold/sub-{subject_label}/anat/sub-{subject_label}_hemi-R_space-T1w_desc-subfields_atlas-bigbrain_dseg.nii.gz"
>>> tmp = nib.load(right_subfield_mask)
p.get_fdata()
# get unique values nonzero values

>>> tmp = tmp.get_fdata()
unique_values = np.unique(tmp[tmp != 0])
print(unique_values)
# print unique values

if len(unique_values) != 8:
    # raise an actual warning using warnings.warn
    # print unique values
    print(unique_values)
    warnings.warn("WARNING: unique values in right_subfield_mask is not equal to 8")

right_subfield_labels = [ label + "_right" for label in subfield_labels]
right_subfield_labels = [right_subfield_labels[i-1] for i in unique_values.astype(int)]
# Instantiate the masker with label image and label values

masker = NiftiLabelsMasker(
    right_subfield_mask,
    labels=right_subfield_labels,
    standardize="zscore_sample",
)

right_subfield_signals = masker.fit_transform(betamap)
print(right_subfield_signals)
>>> # get unique values nonzero values
>>>
>>> unique_values = np.unique(tmp[tmp != 0])
>>> print(unique_values)
**[1. 2. 3. 4. 5. 6. 7. 8.]**
>>> # print unique values
>>>
>>> if len(unique_values) != 8:
...     # raise an actual warning using warnings.warn
...     # print unique values
...     print(unique_values)
...     warnings.warn("WARNING: unique values in right_subfield_mask is not equal to 8")
...
>>> right_subfield_labels = [ label + "_right" for label in subfield_labels]
>>> right_subfield_labels = [right_subfield_labels[i-1] for i in unique_values.astype(int)]
>>> # Instantiate the masker with label image and label values
>>>
>>> masker = NiftiLabelsMasker(
...     right_subfield_mask,
...     labels=right_subfield_labels,
...     standardize="zscore_sample",
... )
>>>
>>> right_subfield_signals = masker.fit_transform(betamap)
>>> print(right_subfield_signals)
**[[-0.52435863 -0.40006825 -0.34511584 -0.5896911  -0.44941693 -0.23059142
  -0.57147044]]**

```

**Comment by bthirion** [comment]
Thx for reporting.
Can you provide the mask and betamap in some way ?

**Comment by ymzayek** [comment]
@andreifoldes I've seen this happen before due to resampling. The labels masker by default resamples the labels_img to the data and this could result in small regions getting wiped out. See `resampling_target` paramter of [NiftiLabelsMasker](https://nilearn.github.io/stable/modules/generated/nilearn.maskers.NiftiLabelsMasker.html). We could add a warning when this happens. WDYT @bthirion ?

**Comment by bthirion** [comment]
Isn't it addressed by #3761

**Comment by ymzayek** [comment]
No explicitly but yes. These [new returns ](https://github.com/nilearn/nilearn/pull/3761/files#diff-96b98185562ed6ed01e5914148fe1df9f6682572ebacf79d91a8b3803aef6cedR118-R143) will help the user map the output back to the input

**Comment by andreifoldes** [comment]
Yes, nice catch, that's probably it! A warning raised would be nice!

**Comment by andreifoldes** [comment]
Is sending the files still necessary?

**Comment by bthirion** [comment]
If you still don't understand what's going on, it's a goo idea to share the files. Best,

**Comment by ymzayek** [comment]
@andreifoldes actually a warning should be raised, see https://github.com/nilearn/nilearn/blob/e4dc5ad0c1aa44dae5f102a5acdad310dc242527/nilearn/maskers/nifti_labels_masker.py#L562-L567

Can you double check if that is raised?

If that's not the case then sending the files or providing a minimal reproducible example to raise the same error might be necessary

**Comment by andreifoldes** [comment]
Yes, will check this week and also send data!
